echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+            weight_aged_sex_adult = ( 'weight_adult' , 'sum' )
+        )
+        .reset_index( )
+    )
+
+    ### Calculate the weight proportions
+    # ---- Merge `aged_sex_weights` with `weight_strata` to get strata weights
+    aged_sex_proportions = aged_sex_weights.merge( aged_proportions ,
+                                                   on = [ 'stratum_num' ] ,
+                                                   how = 'left' )
+
+    # ---- Proportions for all fish
+    aged_sex_proportions[ 'proportion_weight_all' ] = (
+        aged_sex_proportions.weight_aged_sex_all / aged_sex_proportions.weight_stratum_all
+    )
+
+    # ---- Proportions for adult fish
+    aged_sex_proportions[ 'proportion_weight_adult' ] = (
+        aged_sex_proportions.weight_aged_sex_adult / aged_sex_proportions.weight_stratum_all
+    )
+
+    # ---- Fill empty/NaN values with 0's and drop unused columns
+    aged_sex_proportions = aged_sex_proportions.fillna( 0 ).filter( regex = "^(?=proportion_aged_|proportion_weight|stratum_num|sex).*" )
+
+    ### Return output 
+    return aged_sex_proportions.reset_index( drop = True )
+
+def distribute_aged_weight_proportions( proportions_weight_length_age_sex: pd.DataFrame ,
+                                        aged_sex_proportions: pd.DataFrame ):
+    """
+    Distribute overall weight proportions across each sex and age/length bins   
+
+    Parameters
+    ----------
+    proportions_weight_length_age_sex: pd.DataFrame
+        Dataframe containing sexed weight proportions distributed across
+        each age and length bin
+    aged_sex_proportions: pd.DataFrame
+        Dataframe contained weight proportions of sexed fish for aged fish
+    """     
+
+    ### Calculate the normalized age-length-sex aged fish weight proportions
+    # ---- Merge `proportions_weight_length_age_sex` with `aged_sex_proportions`
+    distributed_aged_weight_proportions = proportions_weight_length_age_sex.merge( aged_sex_proportions ,
+                                                                                   on = [ 'stratum_num' , 'sex' ] , 
+                                                                                   how = 'left' )
     
-    ### Normalize the age-length-sex indexed proportions
-    dist_weight_sum = ( 
-       weight_length_age_sex_stratum 
-        .merge( weight_strata , on = [ 'stratum_num' ] ) 
-        .groupby( [ 'stratum_num' , 'sex' ] ) 
-        .apply( lambda df: pd.Series( {  
-            'proportion_normalized': ( df.proportion_weight_all * ( df.summed_weight_all / df.weight_stratum_total ).sum( ) ).sum( ) 
-        } ) ) 
-        .reset_index( ) 
-    ) 
+    # ---- Normalized weight proportions for all fish
+    distributed_aged_weight_proportions[ 'normalized_proportion_weight_all' ] = (
+        distributed_aged_weight_proportions.proportion_weight_sex_all * distributed_aged_weight_proportions.proportion_weight_all
+    )
```

Comment:
Here it would be good to write out what `proportion_weight_sex_all` and `proportion_weight_all` contains, because the variable names themselves do not contain that info.